google/calendar/tools/helper.py

```diff
@@ -0,0 +1,66 @@
+import sys
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+import os
+import logging
+
+
+def setup_logger(name, tool_name: str = "Google Calendar Tool"):
+    """Setup a logger that writes to sys.stderr. This will eventually show up in GPTScript's debugging logs.
+
+    Args:
+        name (str): The name of the logger.
+
+    Returns:
+        logging.Logger: The logger.
+    """
+    # Create a logger
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.DEBUG)  # Set the logging level
+
+    # Create a stream handler that writes to sys.stderr
+    stderr_handler = logging.StreamHandler(sys.stderr)
+
+    # Create a log formatter
+    formatter = logging.Formatter(
+        f"[{tool_name} Debugging Log]: %(asctime)s - %(name)s - %(levelname)s - %(message)s"
+    )
+    stderr_handler.setFormatter(formatter)
+
+    # Add the handler to the logger
+    logger.addHandler(stderr_handler)
+
+    return logger
+
+
+logger = setup_logger(__name__)
+
+
+def get_client(service_name: str = "calendar", version: str = "v3"):
+    token = os.getenv("GOOGLE_OAUTH_TOKEN")
+    if token is None:
+        raise ValueError("GOOGLE_OAUTH_TOKEN environment variable is not set")
+
+    creds = Credentials(token=token)
+    try:
+        service = build(serviceName=service_name, version=version, credentials=creds)
+        return service
+    except HttpError as err:
+        print(err)
+        exit(1)
+
+
+def get_user_timezone(service):
+    """Fetches the authenticated user's time zone from calendar settings."""
+    try:
+        settings = service.settings().get(setting="timezone").execute()
+        return settings.get("value", "UTC")  # Default to UTC if not found
```

Comment:
Does it make sense to default to UTC, or to use `OBOT_USER_TIMEZONE` to get whatever timezone their browser is reporting instead?

Reply:
yeah I'd prefer `OBOT_USER_TIMEZONE` if it's available. is it something I can read from env variable?

Reply:
yes its available as an env var. here is an example - https://github.com/obot-platform/tools/blob/main/google/search-console/src/keywords.js#L4

Reply:
updated to OBOT_USER_TIMEZONE